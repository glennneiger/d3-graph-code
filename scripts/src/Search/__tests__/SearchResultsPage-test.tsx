import React from 'react';
import SearchResultsPage from '../SearchResultsPage';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { reducers } from 'src/store/index';
import renderer from 'react-test-renderer';
import history from "src/history";
import ConnectedRouter from 'react-router-redux/ConnectedRouter';

const store = createStore(
  reducers,
  applyMiddleware(thunk),
);

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <SearchResultsPage />  
        </ConnectedRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});