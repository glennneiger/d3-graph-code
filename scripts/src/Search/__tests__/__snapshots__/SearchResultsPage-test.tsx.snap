// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-12 col-sm-7 offset-sm-1"
  >
    Select a specific tab to narrow your search results.
    <ul
      className="nav nav-tabs"
      role="tablist"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link active"
          href="/results?query=&start=0&filter="
          onClick={[Function]}
        >
          All
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/results?query=&start=0&filter=regulation_number"
          onClick={[Function]}
        >
          Regulation Numbers
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/results?query=&start=0&filter=product_code"
          onClick={[Function]}
        >
          Product Codes
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/results?query=&start=0&filter=application"
          onClick={[Function]}
        >
          Premarket Applications
        </a>
      </li>
    </ul>
    <div
      className="row"
    >
      <div
        className="sk-fading-circle-container"
      >
        <div
          className="sk-fading-circle"
        >
          <div
            className="sk-circle1 sk-circle"
          />
          <div
            className="sk-circle2 sk-circle"
          />
          <div
            className="sk-circle3 sk-circle"
          />
          <div
            className="sk-circle4 sk-circle"
          />
          <div
            className="sk-circle5 sk-circle"
          />
          <div
            className="sk-circle6 sk-circle"
          />
          <div
            className="sk-circle7 sk-circle"
          />
          <div
            className="sk-circle8 sk-circle"
          />
          <div
            className="sk-circle9 sk-circle"
          />
          <div
            className="sk-circle10 sk-circle"
          />
          <div
            className="sk-circle11 sk-circle"
          />
          <div
            className="sk-circle12 sk-circle"
          />
        </div>
      </div>
    </div>
    <div
      className="nova-logged-out-blur"
      onClick={[Function]}
      role="button"
    >
      <p
        className="nova-logged-out-blur-text"
      >
        Please login or sign up for an account to see the results.
        <br />
        By logging in, you will help us improve our service!
        <button
          className="btn btn-outline-primary btn-lg mt-4"
          style={
            Object {
              "display": "block",
              "margin": "auto",
            }
          }
        >
          Login
        </button>
      </p>
    </div>
  </div>
  <div
    className="col-sm-4 nova-search-sidebar-container"
  >
    <div
      className="list-group"
    >
      <div
        className="list-group-item flex-column align-items-start"
      >
        <div
          className="d-flex w-100 justify-content-between"
        >
          <h5
            className="mb-1"
          >
            Filters
          </h5>
          <small />
        </div>
        <div
          className="mb-1"
        >
          <div>
            <div
              className="nova-list-group-sub-item flex-column align-items-start"
            >
              <h6>
                General
              </h6>
              <div
                className="mb-1"
              >
                <div
                  className="nova-filters-left-right"
                >
                  <div
                    className="nova-filters-label"
                  >
                    Device Class
                  </div>
                  <div
                    className="nova-filters-input-wrapper"
                  >
                    <span
                      className="nova-filters-input"
                    >
                      <div
                        className="btn-group btn-group-sm"
                        data-toggle="buttons"
                      >
                        <label
                          className="btn btn-default"
                          disabled={false}
                          type="button"
                        >
                          <input
                            autoComplete="off"
                            checked={false}
                            disabled={false}
                            onChange={[Function]}
                            type="checkbox"
                            value="1"
                          />
                          1
                        </label>
                        <label
                          className="btn btn-default"
                          disabled={false}
                          type="button"
                        >
                          <input
                            autoComplete="off"
                            checked={false}
                            disabled={false}
                            onChange={[Function]}
                            type="checkbox"
                            value="2"
                          />
                          2
                        </label>
                        <label
                          className="btn btn-default"
                          disabled={false}
                          type="button"
                        >
                          <input
                            autoComplete="off"
                            checked={false}
                            disabled={false}
                            onChange={[Function]}
                            type="checkbox"
                            value="3"
                          />
                          3
                        </label>
                        <label
                          className="btn btn-default"
                          disabled={false}
                          type="button"
                        >
                          <input
                            autoComplete="off"
                            checked={false}
                            disabled={false}
                            onChange={[Function]}
                            type="checkbox"
                            value="U"
                          />
                          U
                        </label>
                        <label
                          className="btn btn-default"
                          disabled={false}
                          type="button"
                        >
                          <input
                            autoComplete="off"
                            checked={false}
                            disabled={false}
                            onChange={[Function]}
                            type="checkbox"
                            value="N"
                          />
                          N
                        </label>
                        <label
                          className="btn btn-default"
                          disabled={false}
                          type="button"
                        >
                          <input
                            autoComplete="off"
                            checked={false}
                            disabled={false}
                            onChange={[Function]}
                            type="checkbox"
                            value="f"
                          />
                          f
                        </label>
                      </div>
                    </span>
                    <span
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    />
                  </div>
                  <div
                    className="nova-filters-left-right-clearfix"
                  />
                </div>
                <div
                  className="nova-filters-left-right"
                >
                  <div
                    className="nova-filters-label"
                  >
                    Medical Specialty
                  </div>
                  <div
                    className="nova-filters-input-wrapper"
                  >
                    <span
                      className="nova-filters-input"
                    >
                      <div
                        className="Select is-clearable is-searchable Select--single"
                      >
                        <div
                          className="Select-control"
                          onKeyDown={[Function]}
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                        >
                          <span
                            className="Select-multi-value-wrapper"
                            id="react-select-2--value"
                          >
                            <div
                              className="Select-placeholder"
                            >
                              Select...
                            </div>
                            <div
                              className="Select-input"
                              style={
                                Object {
                                  "display": "inline-block",
                                }
                              }
                            >
                              <input
                                aria-activedescendant="react-select-2--value"
                                aria-expanded="false"
                                aria-haspopup="false"
                                aria-owns=""
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                required={false}
                                role="combobox"
                                style={
                                  Object {
                                    "boxSizing": "content-box",
                                    "width": "5px",
                                  }
                                }
                                value=""
                              />
                              <div
                                style={
                                  Object {
                                    "height": 0,
                                    "left": 0,
                                    "overflow": "scroll",
                                    "position": "absolute",
                                    "top": 0,
                                    "visibility": "hidden",
                                    "whiteSpace": "pre",
                                  }
                                }
                              >
                                
                              </div>
                            </div>
                          </span>
                          <span
                            className="Select-arrow-zone"
                            onMouseDown={[Function]}
                          >
                            <span
                              className="Select-arrow"
                              onMouseDown={[Function]}
                            />
                          </span>
                        </div>
                      </div>
                    </span>
                    <span
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    />
                  </div>
                  <div
                    className="nova-filters-left-right-clearfix"
                  />
                </div>
              </div>
            </div>
            <div
              className="nova-list-group-sub-item flex-column align-items-start"
            >
              <h6>
                Premarket Applications
              </h6>
              <div
                className="mb-1"
              >
                <div
                  className="nova-filters-left-right"
                >
                  <div
                    className="nova-filters-label"
                  >
                    Submission Year
                  </div>
                  <div
                    className="nova-filters-input-wrapper"
                  >
                    <span
                      className="nova-filters-input"
                    >
                      <div
                        className="rc-slider"
                        onBlur={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onMouseDown={[Function]}
                        onTouchStart={[Function]}
                      >
                        <div
                          className="rc-slider-rail"
                          style={Object {}}
                        />
                        <div
                          className="rc-slider-track rc-slider-track-1"
                          style={
                            Object {
                              "left": "0%",
                              "width": "100%",
                            }
                          }
                        />
                        <div
                          className="rc-slider-step"
                        />
                        <div
                          aria-disabled={false}
                          aria-valuemax={2018}
                          aria-valuemin={1980}
                          aria-valuenow={1980}
                          className="rc-slider-handle rc-slider-handle-1"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          role="slider"
                          style={
                            Object {
                              "left": "0%",
                            }
                          }
                          tabIndex={0}
                        />
                        <div
                          aria-disabled={false}
                          aria-valuemax={2018}
                          aria-valuemin={1980}
                          aria-valuenow={2018}
                          className="rc-slider-handle rc-slider-handle-2"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          role="slider"
                          style={
                            Object {
                              "left": "100%",
                            }
                          }
                          tabIndex={0}
                        />
                        <div
                          className="rc-slider-mark"
                        />
                      </div>
                    </span>
                    <span
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    />
                  </div>
                  <div
                    className="nova-filters-left-right-clearfix"
                  />
                </div>
                <div
                  className="nova-filters-left-right"
                >
                  <div
                    className="nova-filters-label"
                  >
                    Application Type
                  </div>
                  <div
                    className="nova-filters-input-wrapper"
                  >
                    <span
                      className="nova-filters-input"
                    >
                      <div
                        className="btn-group btn-group-sm"
                        data-toggle="buttons"
                      >
                        <label
                          className="btn btn-default"
                          disabled={false}
                          type="button"
                        >
                          <input
                            autoComplete="off"
                            checked={false}
                            disabled={false}
                            onChange={[Function]}
                            type="checkbox"
                            value="510k"
                          />
                          510(k)
                        </label>
                        <label
                          className="btn btn-default"
                          disabled={false}
                          type="button"
                        >
                          <input
                            autoComplete="off"
                            checked={false}
                            disabled={false}
                            onChange={[Function]}
                            type="checkbox"
                            value="pma"
                          />
                          PMA
                        </label>
                      </div>
                    </span>
                    <span
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    />
                  </div>
                  <div
                    className="nova-filters-left-right-clearfix"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <small />
      </div>
      <div
        className="list-group-item flex-column align-items-start"
      >
        <div
          className="d-flex w-100 justify-content-between"
        >
          <h5
            className="mb-1"
          >
            About Us
          </h5>
          <small />
        </div>
        <div
          className="mb-1"
        >
          <div>
            These results reflect regulatory concepts available through the FDA website and the 
            <a
              href="https://open.fda.gov/"
              target="_blank"
            >
              OpenFDA API
            </a>
            .
          </div>
        </div>
        <small />
      </div>
      <div
        className="list-group-item flex-column align-items-start"
      >
        <div
          className="d-flex w-100 justify-content-between"
        >
          <h5
            className="mb-1"
          >
            Give Feedback
          </h5>
          <small />
        </div>
        <div
          className="mb-1"
        >
          <div>
            We would love your feedback on our site so far.
            <br />
            Please contact us at 
            <a
              href="mailto:feedback@anora.ai"
            >
              feedback@anora.ai 
            </a>
            for any inquiries or feedback you may have, or submit feedback directly through our form below. Thank you!
            <br />
            <br />
            Some examples of helpful feedback:
            <br />
            <ul>
              <li>
                When I search X I should be seeing Y, but I can't find it.
              </li>
              <li>
                I would love to see a visualization of X, that would be really helpful
              </li>
              <li>
                Would it be possible to show data from X database?
              </li>
              <li>
                You guys are the best.
              </li>
            </ul>
          </div>
          <div
            className="sv_bootstrap_css"
          >
            <div
              className="sv_custom_header"
            />
            <div
              className="sv_container"
            >
              <div
                className="panel-body card-block mt-4"
                id="sp_100"
              >
                <div
                  className=""
                >
                  <div
                    className="sv_row"
                  >
                    <div
                      className="sv_qstn"
                      id="sq_100"
                      style={
                        Object {
                          "width": "100.000000%",
                        }
                      }
                    >
                      <div
                        className={null}
                      />
                      <div
                        className={null}
                      >
                        <textarea
                          aria-label="* comment"
                          className="form-control"
                          cols={50}
                          id="sq_100i"
                          maxLength={null}
                          onBlur={[Function]}
                          onChange={[Function]}
                          placeholder="Do you have any comments about how we can improve Nova Search?"
                          readOnly={false}
                          rows={4}
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="sv_row"
                  >
                    <div
                      className="sv_qstn"
                      id="sq_101"
                      style={
                        Object {
                          "width": "100.000000%",
                        }
                      }
                    >
                      <div
                        className={null}
                      />
                      <div
                        className={null}
                      >
                        <input
                          aria-label="email"
                          className="form-control"
                          id="sq_101i"
                          maxLength={null}
                          onBlur={[Function]}
                          onChange={[Function]}
                          placeholder="Your email (optional)"
                          size={25}
                          type="email"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="panel-footer card-footer"
                >
                  <input
                    className="btn btn-green sv_complete_btn"
                    onClick={[Function]}
                    style={
                      Object {
                        "marginRight": "5px",
                      }
                    }
                    type="button"
                    value="Submit"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <small />
      </div>
    </div>
  </div>
</div>
`;
